Warehouse.Infrastructure/ProcutsRepository/ProductMockRepository.cs

```diff
@@ -0,0 +1,59 @@
+    new Product {Price = 36.25m, Title = "Suede Ankle Boots", Description = "Trendy suede ankle boots to elevate your footwear collection.", Sizes = new List<string>{"7", "8", "9", "10" } },
+    new Product {Price = 16.50m, Title = "Ribbed Knit Sweater", Description = "Cozy ribbed knit sweater for a warm and stylish winter.", Sizes = new List<string>{"Medium", "Large", "X-Large" } },
+    new Product {Price = 27.99m, Title = "Printed Beach Towel", Description = "Vibrantly printed beach towel for a stylish day at the beach.", Sizes = new List<string>{"One Size" } },
+    new Product {Price = 49.50m, Title = "Formal Dress Shirt", Description = "Elegant formal dress shirt for a polished and sophisticated look.", Sizes = new List<string>{"15.5", "16", "16.5", "17" } },
+
+    new Product {Price = 18.95m, Title = "Casual Joggers", Description = "Comfortable and casual joggers for a relaxed and sporty style.", Sizes = new List<string>{"Small", "Medium", "Large" } },
+    new Product {Price = 33.75m, Title = "Oversized Sunglasses", Description = "Chic oversized sunglasses to add a touch of glamour to your look.", Sizes = new List<string>{"One Size" } },
+    new Product {Price = 14.49m, Title = "Hooded Sweatshirt Dress", Description = "Stylish hooded sweatshirt dress for a trendy and comfortable outfit.", Sizes = new List<string>{"Small", "Medium", "Large" } },
+    new Product {Price = 26.99m, Title = "Leather Belt", Description = "Classic leather belt to complete your everyday attire.", Sizes = new List<string>{"32", "34", "36", "38" } },
+    new Product {Price = 42.95m, Title = "Tailored Trench Coat", Description = "Sophisticated tailored trench coat for a timeless and polished look.", Sizes = new List<string>{"Medium", "Large", "X-Large" } },
+
+    new Product {Price = 23.99m, Title = "Printed Silk Blouse", Description = "Elegant printed silk blouse for a refined and feminine ensemble.", Sizes = new List<string>{"Small", "Medium", "Large" } },
+    new Product {Price = 39.50m, Title = "Rugged Backpack", Description = "Durable rugged backpack for adventurous outdoor activities.", Sizes = new List<string>{"One Size" } },
+    new Product {Price = 17.25m, Title = "Floral Print Maxi Dress", Description = "Graceful floral print maxi dress for a bohemian and romantic look.", Sizes = new List<string>{"Small", "Medium", "Large" } },
+    new Product {Price = 29.99m, Title = "High-Top Sneakers", Description = "Trendy high-top sneakers for a fashion-forward and casual style.", Sizes = new List<string>{"6", "7", "8", "9", "10" } },
+    new Product {Price = 48.00m, Title = "Wool Fedora Hat", Description = "Timeless wool fedora hat for a classic and sophisticated appearance.", Sizes = new List<string>{"Small", "Medium", "Large" } },
+};
+
+
+    Task<IEnumerable<Product>> IProductRepository.GetAllProducts()
```

Comment:
Try this method to be async it will be a better mock.
The idea is to be closest to the real world.

Reply:
Done.